apps/admin/frontend/src/screens/manual_data_import_precinct_screen.tsx

```diff
@@ -478,51 +568,106 @@ export function ManualDataImportPrecinctScreen(): JSX.Element {
           const contestTally = currentPrecinctTally.contestTallies[contest.id];
           assert(contestTally);
 
+          const contestWriteInCandidates = writeInCandidates.filter(
+            ({ contestId }) => contestId === contest.id
+          );
+          const contestTempWriteInCandidates = tempWriteInCandidates.filter(
+            ({ contestId }) => contestId === contest.id
+          );
+          const disallowedNewWriteInCandidateNames =
+            contest.type === 'candidate'
+              ? [
+                  ...contest.candidates,
+                  ...contestWriteInCandidates,
+                  ...contestTempWriteInCandidates,
+                ].map(({ name }) => normalizeWriteInName(name))
+              : [];
+
           return (
             <ContestData key={contest.id}>
               <Text small>{getContestDistrictName(election, contest)}</Text>
               <h3>{contestTitle}</h3>
               <Table borderTop condensed>
                 <tbody>
-                  {contest.type === 'candidate' &&
-                    getCandidatesFromContestTally(contestTally).map(
-                      (candidate) => (
+                  {contest.type === 'candidate' && (
+                    <React.Fragment>
```

Comment:
This form has a lot of repeated buts but when I tried to consolidate it felt harder to think through.